### PR TITLE
magnum: Update Magnum config template to Newton release

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

File: chef/cookbooks/magnum/templates/default/magnum.conf.erb [hunk 1]
```diff
@@ -666,18 +729,71 @@ region_name = <%= @keystone_settings['endpoint_region'] %>
 # Version of Heat API to use in heatclient. (string value)
 #api_version = 1
 
-# Optional CA cert file to use in SSL connections. (string value)
-#ca_file = <None>
 
-# Optional PEM-formatted certificate chain file. (string value)
-#cert_file = <None>
+[keystone_auth]
```
Comment (toabctl) [inline chef/cookbooks/magnum/templates/default/magnum.conf.erb]: does that work without setting any option in the ```[keystone_auth]``` section?

File: chef/cookbooks/magnum/templates/default/magnum.conf.erb [hunk 2]
```diff
@@ -301,115 +358,110 @@ region_name = <%= @keystone_settings['endpoint_region'] %>
...
+# Certificate Manager plugin. Defaults to barbican. (string value)
+#cert_manager_type = barbican
+cert_manager_type = <%= "#{node[:magnum][:cert][:cert_manager_type]}" %>
 
-# Location of template to build a k8s cluster on CoreOS. (string value)
-#k8s_coreos_template_path = $pybasedir/templates/kubernetes/kubecluster-coreos.yaml
+# Absolute path of the certificate storage directory. Defaults to
+# /var/lib/magnum/certificates/. (string value)
+#storage_path = /var/lib/magnum/certificates/
+storage_path = /var/lib/magnum/certificates/
```
Comment (toabctl) [inline chef/cookbooks/magnum/templates/default/magnum.conf.erb]: not needed. same as default

Comment (mjura) [inline chef/cookbooks/magnum/templates/default/magnum.conf.erb]: Ok, let's keep it default

**Comment by mjura** [comment]
Hmmm, it looks that Travis is failing because of some unrelated issue

```
$ bundle exec rake
rake aborted!
....
```

**Comment by mjura** [comment]
> plan was to install a minified config..

@dirkmueller: sure, we can still do this, I did the work already.

**Comment by mjura** [comment]
@dirkmueller: I've adopted changes to minimal config

**Comment by mjura** [comment]
@dirkmueller: Do you have any further comments?

File: chef/cookbooks/magnum/recipes/common.rb [hunk 3]
```diff
@@ -47,7 +49,11 @@
     bind_host: bind_host,
     bind_port: bind_port,
     sql_connection: sql_connection,
-    rabbit_settings: fetch_rabbitmq_settings,
+    transport_url: "rabbit://#{rabbit_settings[:user]}:"\
```
Comment (dirkmueller) [inline chef/cookbooks/magnum/recipes/common.rb]: useless duplication, please just use rabbitmq_settings[:url] in the config template

Comment (mjura) [inline chef/cookbooks/magnum/recipes/common.rb]: Done

File: chef/cookbooks/magnum/recipes/common.rb [hunk 4]
```diff
@@ -37,7 +37,9 @@
 bind_port = network_settings[:api][:bind_port]
 bind_host = network_settings[:api][:bind_host]
 
-template "/etc/magnum/magnum.conf" do
+rabbit_settings = fetch_rabbitmq_settings
+
+template "/etc/magnum/magnum.conf.d/100-magnum.conf" do
```
Comment (toabctl) [inline chef/cookbooks/magnum/recipes/common.rb]: The other barclamps use a config_file attribute for that. Could you also do that?

Comment (mjura) [inline chef/cookbooks/magnum/recipes/common.rb]: I've added new commit which is introducing this change

**Comment by mjura** [comment]
@toabctl, @dirkmueller: I've addressed all comments

File: chef/cookbooks/magnum/recipes/common.rb [hunk 5]
```diff
@@ -47,7 +49,7 @@
     bind_host: bind_host,
     bind_port: bind_port,
     sql_connection: sql_connection,
-    rabbit_settings: fetch_rabbitmq_settings,
+    transport_url: rabbit_settings[:url],
```
Comment (dirkmueller) [inline chef/cookbooks/magnum/recipes/common.rb]: just use @rabbit_settings[:url] in the template directly..

**Comment by toabctl** [comment]
Timeout while creating the magnum cluster...

**Pull request merged**